Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -2208,6 +2209,36 @@ void WebProcessProxy::getNotifications(const URL& registrationURL, const String&
     WebNotificationManagerProxy::sharedServiceWorkerManager().getNotifications(registrationURL, tag, sessionID(), WTFMove(callback));
 }
 
+void WebProcessProxy::setAppBadge(std::optional<WebPageProxyIdentifier> identifier, const SecurityOriginData& origin, std::optional<uint64_t> badge)
+{
+    if (!identifier) {
+        websiteDataStore()->workerUpdatedAppBadge(origin, badge);
+        return;
+    }
+
+    // This page might have gone away since the WebContent process sent this message,
+    // and that's just fine.
+    auto page = m_pageMap.get(*identifier);
+    if (!page)
+        return;
+
+    page->uiClient().updateAppBadge(*page, badge);
+}
+
+void WebProcessProxy::setClientBadge(std::optional<WebPageProxyIdentifier> identifier, const SecurityOriginData&, std::optional<uint64_t> badge)
```

Comment:
identifier should be named pageID as per the pattern in this file. identifier is too generic.